src/ext/BalExtension/wixstdba/WixStandardBootstrapperApplication.cpp

```diff
@@ -576,6 +566,7 @@ class CWixStandardBootstrapperApplication : public CBalBaseBootstrapperApplicati
         __in HRESULT /*hrStatus*/
         )
     {
+        UpdateCacheProgress(100); // Make sure cache progress got set to 100%.
```

Comment:
It's not always 100% here because we're going off of the progress reported in OnCacheAcquireProgress.  The engine only sends this message while caching a payload.  When planning to cache a payload that is inside of a container, the size of the container is added to the total cache size.  During execution, the container is only acquired if one of its payloads needs to be cached.  So if all of the payloads in a planned container were already cached, at the end of caching the progress will be less than 100%.